src/composition.rs

```diff
@@ -444,7 +449,19 @@ impl<'a> ast::walk::Visitor<'a> for CompositionQueryAnalyzer {
                             _ => {
                                 // Treat these all as tag filters.
                                 if let ast::Expression::StringLit(string_literal) = &binary_expr.right {
-                                    self.tag_values.push((ident.name.clone(), string_literal.value.clone()));
+                                    let mut inserted = false;
+                                    for (i, (tag_key, _)) in self.tag_values.iter().enumerate() {
+                                        if tag_key == &ident.name {
+                                            self.tag_values[i].1.push(string_literal.value.clone());
+                                            inserted = true;
+                                            break;
+                                        }
+                                    }
+                                    if !inserted {
+                                        self.tag_values.push((
+                                            ident.name.clone(), vec![string_literal.value.clone()]
+                                        ));
+                                    }
```

Comment:
I have this pattern through out this file to set/get `self.tag_values`. The reason I chose to use a `for` loop and have a `break` in the loop is to save the number of iterations we have to do on a vec. In this code, there is only one iteration.

I'm new to rust so there might be functions/patterns do the same thing more efficiently and easier to read. I'm open to change to other ways!